server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/mapping/ExtendedSource.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.engine.mapping;
+
+import org.eclipse.lsp.cobol.core.model.tree.CopyNode;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.Range;
+
+import java.util.*;
+
+/** Container fot mappable modifications on source code files */
+public class ExtendedSource {
+
+  private final Map<String, DocumentMap> documents = new HashMap<>();
+  private final String mainUri;
+
+  public ExtendedSource(String text, String uri) {
+    mainUri = uri;
+    documents.put(uri, new DocumentMap(uri, text));
+  }
+
+  @Deprecated
+  public ExtendedSource(TextTransformations textTransformations) {
```

Comment:
Why do we have a deprecated method, in a newly created class?

Reply:
Fixed